gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinDriverIntegrateTest.java

```diff
@@ -1596,6 +1601,18 @@ public void shouldCloseAllClientsOnCloseOfCluster() throws Exception {
         cluster.close();
     }
 
+    @Test
+    public void shouldSendUserAgent() throws Exception {
```

Comment:
Can you please add a unit test for the client pattern, `g.with("userAgent", "test").V()` as well?

Reply:
`userAgent` is not overridable on the traversal level, it's purely a request based thing.
It's not accessible from scripts, and is only there for graph providers who have customised their request handling to pick up this extra info.

Reply:
Hmm, I see you mean bytecode requests. I'll update the code.

Reply:
Added and discovered an issue in DriverRemoteConnection.java in the process.